application/app/src/main/java/uk/nhs/digital/uec/api/repository/elasticsearch/impl/ServiceRepository.java

```diff
@@ -39,80 +39,138 @@ public List<DosService> findServiceBySearchTerms(List<String> searchTerms) {
+    // Adjust number of results to return from ES depending on how many words are in
+    // the search
     int numOfSpaces = StringUtils.countMatches(searchCriteria, " ");
 
     int numServicesToReturnFromEs = apiRequestParams.getMaxNumServicesToReturnFromElasticsearch();
     if (numOfSpaces == 2) {
-      numServicesToReturnFromEs =
-        apiRequestParams.getMaxNumServicesToReturnFromElasticsearch3SearchTerms();
+      numServicesToReturnFromEs = apiRequestParams.getMaxNumServicesToReturnFromElasticsearch3SearchTerms();
     }
     if (numOfSpaces > 2) {
-      numServicesToReturnFromEs =
-        Math.floorDiv(
+      numServicesToReturnFromEs = Math.floorDiv(
           apiRequestParams.getMaxNumServicesToReturnFromElasticsearch3SearchTerms(), 2);
     }
 
     return performSearch(searchCriteria, numServicesToReturnFromEs);
   }
 
   @Override
-  public List<DosService> findServiceByLocation(String searchLocation) throws NotFoundException {
+  public List<DosService> findServiceByPostcode(String searchPostcode) throws NotFoundException {
 
-    log.info("Request Params: {}", searchLocation);
+    log.info("Request Params: {}", searchPostcode);
     log.info("Fuzzy level: {}", apiRequestParams.getFuzzLevel());
     log.info(
-      "Number of services to get from elasticsearch: "
-        + apiRequestParams.getMaxNumServicesToReturnFromElasticsearch());
+        "Number of services to get from elasticsearch: "
+            + apiRequestParams.getMaxNumServicesToReturnFromElasticsearch());
 
-    log.info("Validate Postcode: {}", searchLocation);
+    log.info("Validate Postcode: {}", searchPostcode);
     Pattern pattern = Pattern.compile(POSTCODE_REGEX);
-    if (!pattern.matcher(searchLocation).matches()) {
+    if (!pattern.matcher(searchPostcode).matches()) {
       throw new NotFoundException(ErrorMessageEnum.INVALID_LOCATION.getMessage());
     }
 
     // Get the first part of the postcode
-    String searchCriteria = searchLocation.substring(0, 4).trim();
+    String searchCriteria = searchPostcode.substring(0, 4).trim();
     return performSearch(searchCriteria, null);
   }
 
+  @Override
+  public List<DosService> findServiceByLatitudeAndLongitude(String searchLatitude, String searchLongitude,
+      String distanceRange) throws NotFoundException {
+
+    log.info("Request Params: Lat: {}, Lng: {}", searchLatitude, searchLongitude);
+    log.info("Fuzzy level: {}", apiRequestParams.getFuzzLevel());
+    log.info(
+        "Number of services to get from elasticsearch: "
+            + apiRequestParams.getMaxNumServicesToReturnFromElasticsearch());
+
+    log.info("Validate geo location points: {} {}", searchLatitude, searchLongitude);
+    if (!NumberUtils.isCreatable(searchLatitude) || (!NumberUtils.isCreatable(searchLongitude))) {
+      throw new NotFoundException(ErrorMessageEnum.INVALID_LOCATION.getMessage());
+    }
+
+    return performSearch(searchLatitude, searchLongitude, distanceRange, null);
```

Comment:
we need to pass in apiRequestParams.getMaxNumServicesToReturnFromElasticsearch() instead of null